### ISSUE TITLE
Running on NodeLua

### ISSUE THREAD

**Issue by Bazmundi** [fsantanna/luagravity#1]
What is the likelihood of porting this to something like the ESP8266 which is already running Lua?

**Comment by fsantanna** [comment]
Hi @Bazmundi, 
If `NodeLua` supports coroutines, then `LuaGravity` should run without any problems.
From the docs, it seems `NodeLua` doesn't remove anything from standard Lua.

**Comment by Bazmundi** [comment]
Yes. Thanks for confirming this. I am starting my experiments this weekend.

**Comment by Bazmundi** [comment]
Ah, I chatted to people on the nodemcu site.  Turns out there is a tool for uploading groups of files to root but not in directories and therefore not as files packaged under directories.  What mods do you suggest might be needed to "unpackadize" luagravity distro please?

**Comment by fsantanna** [comment]
To execute `tests/1.lua`, I did the following:

```
> mv init.lua luagravity.lua
> mv env/simple.lua luagravity_env_simple.lua
> mv tests/1.lua 1.lua
> vi 1.lua # change line 2 to "local env = require 'luagravity_env_simple'"
> lua 1.lua
```

**Comment by Bazmundi** [comment]
Cheers, I will try that.

**Comment by Bazmundi** [comment]
Yep, works for windoze but not on ESP-01.

The problem is the ESP8266 crammed with nodeMCU does not appear to have
enough room left to compile the luagravity.lua file.  I am looking into how
I might compile to *.lc offboard and copying it on (assuming that gets me
around the problem)

Thanks for your help with in any event.  I am on nodeMCU list now to try to
progress this.

Cheers,
A

**Comment by Bazmundi** [comment]
Yep. Nope. That is not happening yet.
1. Luagravity not compile against main distro as there was apparently not
   enough memory.
2. nodeMCU no longer in active development so no support.
3. Tried web based build using minimum modules that gave more memory.
4. luagravity still ran out of memory compiling on ESP8266 compiling on
   chip.
5. There is an option in ESPlorer IDE to compile and upload *.lc which I
   tried in vein.
6. A hex dump of luagravity.l http://luagravity.lsc was way too short to
   represent the code size I expected and was mostly zeros.
7. A "require" seemed to find the luagravity but when attempting to use it
   came up global nil.
8. A look at the sdk for nodeMCU was the usual OMG! so complicated and no
   useful help files etc.

In all, a nice little experiment but I have also found other problems with
the nodeMCU on the ESP8266 that likely make it less than ideal to spend
time on creative things rather than debugging.

Thanks again for your help.

I will persist with ceu on the arduino instead.

Cheers,
A
